### PR TITLE
Add portable installers for Linux and Windows

### PR DESCRIPTION
**Pull request by Moguri**

### CODE REVIEW

**Comment by Moguri** [comment]
This replaces PR #89

**Comment by Moguri** [comment]
Some requests from wezu and rdb on IRC:
* Have separate modes for portable and installer
* Make sure DLLs are copied for Windows builds (make use of unused extraFiles)
* Use supplied icon
* Make archiving an option, and don't have a regular folder in the build dir if an archive is requested
* Remove duplicate binary (usr/bin) in Linux build

**Comment by Moguri** [comment]
Adding some more commits to address the issues brought up in the last comment. I did not find a duplicate binary in Linux builds. Also, icons may have to wait until later. The steps used by the Installer for icons (.desktop files on Linux and shortcuts on Windows) do not work with portable builds. On Windows we'll need to edit the resource section of the executable. I'm not sure what we'll do for Linux; there may be some freedesktop spec for this.

File: direct/src/p3d/DeploymentTools.py [hunk 1]
```diff
@@ -701,19 +702,27 @@ def build(self, output, platform = None):
...
-        elif "_" in platform:
-            osname, arch = platform.split("_", 1)
-            if osname == "linux":
-                self.buildDEB(output, platform)
-                self.buildArch(output, platform)
-                return
-            elif osname == "osx":
+        if portable:
+            if platform.startswith("osx"):
                 self.buildPKG(output, platform)
```
Comment (rdb) [inline direct/src/p3d/DeploymentTools.py]: Should be buildAPP

**Comment by Moguri** [comment]
Adding BAM file and Python script to test the changes.

[bulletbamtest.zip](https://github.com/panda3d/panda3d/files/355661/bulletbamtest.zip)

**Comment by Moguri** [comment]
I am abandoning this PR in favor of deploy-ng.

**Pull request closed without merging**